code/modules/flufftext/Hallucination.dm

```diff
@@ -1168,20 +1323,33 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 		else
 			fakemob = target //ever been so lonely you had to haunt yourself?
 		if(fakemob)
-			sleep(rand(20, 50))
-			to_chat(target, "<span class='deadsay'><b>DEAD: [fakemob.name]</b> says, \"[pick("rip","why did i just drop dead?","hey [target.first_name()]","git gud","you too?","is the AI rogue?",\
-				"i[prob(50)?" fucking":""] hate [pick("blood cult", "clock cult", "revenants", "this round","this","myself","admins","you")]")]\"</span>")
-	sleep(rand(70,90))
-	target.set_screwyhud(SCREWYHUD_NONE)
-	target.SetParalyzed(0)
-	target.silent = FALSE
+			delay = rand(20, 50)
+			addtimer(CALLBACK(GLOBAL_PROC, .proc/to_chat, target, "<span class='deadsay'><b>DEAD: [fakemob.name]</b> says, \"[pick("rip","why did i just drop dead?","hey [target.first_name()]","git gud","you too?","is the AI rogue?",\
+				"i[prob(50)?" fucking":""] hate [pick("blood cult", "clock cult", "revenants", "this round","this","myself","admins","you")]")]\"</span>"), delay)
+
+	addtimer(CALLBACK(src, .proc/cleanup), delay + rand(70, 90))
```

Comment:
This is more time then by rights it should take, if you want to maintain behavior just do 70, 90, if you want to give it a static delay give it a 20

Reply:
I'm confused, how does what I have now now maintain the current behavior?

Reply:
it currently fires in 2 to 5 seconds, and deletes itself after 7 to 9 seconds, with your current changes it would fire in the same manner, but delete itself in 9 to 14 seconds.

Reply:
I think you're wrong.

The current code rolls a chance to make someone talk in dead chat. If this check passes, it'll sleep for 2 to 5 seconds. Then, it sleeps 7 to 9 seconds after showing the chat message.

Reply:
ah excuse me I misread the code, you're right.